agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -344,3 +356,345 @@ func TestTwoTasksSharedLocalVolume(t *testing.T) {
+		Name:        aws.String(parameterName),
+		Value:       aws.String(testCredSpecData),
+		Type:        aws.String("String"),
+	}
+
+	// create parameter in parameter store if it does not exist
+	_, err := ssmClient.PutParameter(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case ssm.ErrCodeParameterAlreadyExists:
+				t.Logf("Parameter %v already exists in SSM Parameter Store", parameterName)
+				break
+			default:
+				require.NoError(t, err, "SSM PutParameter call failed")
+			}
+		}
+	}
+
+	// read parameter back to obtain ARN
+	paramInput := &ssm.GetParameterInput{
+		Name: aws.String(parameterName),
+	}
+	data, err := ssmClient.GetParameter(paramInput)
+	require.NoError(t, err)
+
+	// Obtain ARN of SSM param
+	gmSAParameterARN := *data.Parameter.ARN
+
+	// Setup agent
+	RequireDockerAPIVersion(t, ">=1.40")
+
+	agentOptions := AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION": "true",
+			"ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION":                  "1m",
+		},
+	}
+
+	agent := RunAgent(t, &agentOptions)
+	defer agent.Cleanup()
+
+	agent.RequireVersion(">=1.33.0")
+
+	// Setup taskdef
+	tdOverrides := make(map[string]string)
+	tdOverrides["$$$TEST_REGION$$$"] = aws.StringValue(ECS.Config.Region)
+	tdOverrides["$$$EXECUTION_ROLE$$$"] = executionRole
+	tdOverrides["$$$TEST_SSM_ARN$$$"] = gmSAParameterARN
+
+	testTask, err := agent.StartTaskWithTaskDefinitionOverrides(t, "gmsa-ssm", tdOverrides)
+	require.NoError(t, err)
+
+	// Wait for the container to start
+	testTask.WaitRunning(waitTaskStateChangeDuration)
+	taskID, err := GetTaskID(aws.StringValue(testTask.TaskArn))
+	require.NoError(t, err)
+	assert.NotEmpty(t, taskID)
+
+	// Verify container metadata
+	cid, err := agent.ResolveTaskDockerID(testTask, "windows_sample_app")
+	require.NoError(t, err, "Error resolving docker id for container in task")
+	assert.NotEmpty(t, cid)
+
+	// Setup docker client to validate credentialspec
+	endpoint := utils.DefaultIfBlank(os.Getenv(dockerEndpointEnvVariable), dockerEndpoint)
+	client, _ := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+
+	// credentialspec=file://ssm_c3ce6880-f815-4044-af03-29f934836293_FunctionalTest-gMSA-SSM
+	testCredentialSpecOption := fmt.Sprintf("credentialspec=file://ssm_%s_%s", taskID, parameterName)
+
+	err = verifyContainerCredentialSpec(client, cid, testCredentialSpecOption)
+	assert.NoError(t, err)
+
+	// Stop test task
+	testTask.Stop()
+
+	err = testTask.WaitStopped(waitTaskStateChangeDuration)
+	assert.NoError(t, err)
+
+	// Cleanup the test file
+	testCredSpecFilePath := fmt.Sprintf("C:/ProgramData/docker/credentialspecs/ssm_%s_%s", taskID, parameterName)
+	err = os.Remove(testCredSpecFilePath)
+	assert.NoError(t, err)
+
+	// Cleanup the ssm parameter
```

Comment:
You don't really need to clean up the parameter, since we will keep using it, you can just leave it there.

Reply:
It just to make sure that each test validates the entire workflow. This is harmless.